inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/hook/MethodHookGenerator.java

```diff
@@ -122,14 +123,45 @@ private void addDataProviderCalls(MethodReflectionInformation method, MethodHook
      * @param providerCallConfig the specification of the call to the data provider
      * @return the executable data provider
      */
-    private BoundDataProvider generateAndBindDataProvider(MethodReflectionInformation methodInfo, String dataKey, DataProviderCallConfig providerCallConfig) {
+    private IHookAction generateAndBindDataProvider(MethodReflectionInformation methodInfo, String dataKey, DataProviderCallConfig providerCallConfig) {
         GenericDataProviderConfig providerConfig = providerCallConfig.getProvider();
+        val callSettings = providerCallConfig.getCallSettings();
         val injectedProviderClass = dataProviderGenerator.getOrGenerateDataProvider(providerConfig, methodInfo.getDeclaringClass());
 
         val dynamicAssignments = getDynamicInputAssignments(methodInfo, providerCallConfig);
         val constantAssignments = getConstantInputAssignments(methodInfo, providerCallConfig);
 
-        return BoundDataProvider.bind(dataKey, providerConfig, injectedProviderClass, constantAssignments, dynamicAssignments);
+        IHookAction providerCall = BoundDataProvider.bind(dataKey, providerConfig, injectedProviderClass, constantAssignments, dynamicAssignments);
+
+        return addConditionsToProviderCall(callSettings, providerCall);
+    }
+
+    private IHookAction addConditionsToProviderCall(DataProviderCallSettings callSettings, IHookAction providerCall) {
+        if (!StringUtils.isEmpty(callSettings.getOnlyIfTrue())) {
+            String conditionDataKey = callSettings.getOnlyIfTrue();
+            providerCall = new ConditionalHookAction((e) -> {
+                Object val = e.getInspectitContext().getData(conditionDataKey);
+                return val != null && (Boolean) val;
```

Comment:
If the data key is referencing something which is not a boolean it will throw an exception. Is this by purpose of should we use `Boolean.valueOf(..)`? This will not throw an exception but has less transparency why a value might be false all the time?.

Reply:
Boolean.valueOf only accepts "Strings", whereas the value could be any type off Object if misused.
I on purpose only allowed "Booleans".
When this is not a boolean, an exception will be raised. This exception is caught by the MethodHook, which prints out an error to the log and then disabled the erronous data provider.